hs_collection_resource/views.py

```diff
@@ -104,3 +111,151 @@ def update_collection_for_deleted_resources(request, shortkey, *args, **kwargs):
     finally:
         return JsonResponse(ajax_response_data)
 
+
+def update_collection_coverages(request, shortkey, *args, **kwargs):
+
+    ajax_response_data = {'status': "success"}
+    try:
+        collection_res, is_authorized, user = authorize(request, shortkey,
+                                                        needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+
+        if collection_res.resource_type.lower() != "collectionresource":
+            raise Exception("Resource {0} is not a collection resource.".format(shortkey))
+
+        new_coverage_list = _update_collection_coverages(collection_res)
+        ajax_response_data['new_coverage_list'] = new_coverage_list
+
+        resource_modified(collection_res, user)
+
+    except Exception as ex:
+        logger.error("Failed to update collection coverages. Collection resource ID: {0}. "
+                     "Error:{1} ".format(shortkey, ex.message))
+
+        ajax_response_data = {'status': "error", 'message': ex.message}
+    finally:
+        return JsonResponse(ajax_response_data)
+
+
+def calculate_collection_coverages(request, shortkey, *args, **kwargs):
+
+    ajax_response_data = {'status': "success"}
+    try:
+        collection_res, is_authorized, user = authorize(request, shortkey,
+                                                        needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE)
+
+        if collection_res.resource_type.lower() != "collectionresource":
+            raise Exception("Resource {0} is not a collection resource.".format(shortkey))
+
+        new_coverage_list = _calculate_collection_coverages(collection_res)
+        ajax_response_data['new_coverage_list'] = new_coverage_list
+
+        resource_modified(collection_res, user)
+
+    except Exception as ex:
+        logger.error("Failed to calculate collection coverages. Collection resource ID: {0}. "
+                     "Error:{1} ".format(shortkey, ex.message))
+
+        ajax_response_data = {'status': "error", 'message': ex.message}
+    finally:
+        return JsonResponse(ajax_response_data)
+
+
+def _update_collection_coverages(collection_res_obj):
+    """
+    update the collection coverages metadata.
+    :param collection_res_obj: collection obj
```

Comment:
A new reader may not understand what's a ```collection obj```. May be better to say an 'instance of CollectionResource type'